src/Components/Onboarding/Hooks/useOnboardingContext.tsx

```diff
@@ -73,33 +75,49 @@ const OnboardingContext = createContext<{
   current: string
   dispatch: React.Dispatch<Action>
   next(): void
-  onDone(): void
+  onComplete(): void
+  onClose(): void
   progress: number
   state: State
   workflowEngine: WorkflowEngine
 }>({
   current: "",
   dispatch: () => {},
   next: () => {},
-  onDone: () => {},
+  onComplete: () => {},
+  onClose: () => {},
   progress: 0,
   state: DEFAULT_STATE,
   workflowEngine: new WorkflowEngine({ workflow: [] }),
 })
 
 interface OnboardingProviderProps {
-  onDone(): void
+  onClose(): void
 }
 
 export const OnboardingProvider: FC<OnboardingProviderProps> = ({
   children,
-  onDone,
+  onClose,
 }) => {
   const basis = useRef<State>(DEFAULT_STATE)
+  const { relayEnvironment } = useSystemContext()
+  const { submitUpdateMyUserProfile } = useUpdateMyUserProfile({
+    relayEnvironment,
+  })
+
+  const handleComplete = async () => {
+    try {
+      await submitUpdateMyUserProfile({
+        completedOnboarding: true,
+      })
```

Comment:
this is the only relevant addition

Reply:
oh ok well then disregard any other comments then for sure

Reply:
am i correct in seeing that this function handles the mutation to update the user profile? not the collector profile??

Reply:
This mutation will eventually call [this line](https://github.com/artsy/gravity/blob/fec1f717d96a144c46e88aa66731bde8403c01a8/app/api/v1/me_endpoint.rb#L69); in this case we're just re-wiring-back-up what existed before.